tests/integration_test.py

```diff
@@ -0,0 +1,82 @@
+import unittest
+
+from model import SUPPORTED_SUMM_MODELS
+from model.base_model import SummModel
+from evaluation import SUPPORTED_EVALUATION_METRICS
+from evaluation.base_metric import SummMetric
+from dataset import SUPPORTED_SUMM_DATASETS
+from dataset.st_dataset import SummInstance, SummDataset
+
+import random
+import time
+from typing import Dict, List, Union
+
+from evaluation.rouge_metric import Rouge
+from evaluation.rougewe_metric import RougeWe
+from model.single_doc import LexRankModel
+from dataset.non_huggingface_datasets import ScisummnetDataset
+from dataset.huggingface_datasets import CnndmDataset
+
+class IntegrationTests(unittest.TestCase):
+    @staticmethod
+    def print_with_color(s: str, color: str):
+        print(f"\033[{color}m{s}\033[0m")
+
+    @staticmethod
+    def flatten_list_to_str(doc: Union[str, List[str]]) -> str:
+        return " ".join(doc) if isinstance(doc, list) else doc
+
+    def retrieve_test_instances(self, dataset_instances: List[SummInstance], num_instances = 5) -> List[SummInstance]:
+        test_instances = []
+        for i in range(num_instances):
+            test_instances.append(dataset_instances[random.randint(0, len(dataset_instances) - 1)])
+        return test_instances
+
+    def _test_single_integration(self, dataset: SummDataset, test_instances: List[SummInstance], model: SummModel, metric: SummMetric):
+        IntegrationTests.print_with_color(f"{'#' * 20} Testing: {dataset.dataset_name} dataset, {model.model_name} model, {metric.metric_name} evaluation metric {'#' * 20}", "35")
+        # if dataset.is_multi_document != model.is_multi_document or dataset.is_dialogue_based != model.is_dialogue_based or dataset.is_query_based != dataset.is_query_based:
+        #     print("Skipping because summarization tasks of dataset and model don't match\n")
+        #     return
+        src = [ins.source[0] for ins in test_instances] if isinstance(dataset, ScisummnetDataset) else [IntegrationTests.flatten_list_to_str(ins.source) for ins in test_instances]
+        tgt = [ins.summary for ins in test_instances]
+        prediction = model.summarize([src] if model.is_multi_document else src)
+        print(prediction)
+        print(tgt)
+        score_dict = metric.evaluate(prediction, [" ".join(tgt)] if model.is_multi_document else tgt)
+        print(score_dict)
+
+    def test_all(self):
+        IntegrationTests.print_with_color("Initializing all datasets...", "35")
+        # datasets = []
+        # for dataset_cls in SUPPORTED_SUMM_DATASETS:
+        #     print(dataset_cls)
+        #     ds = dataset_cls()
+        #     datasets.append(ds)
+        lxr_dataset = CnndmDataset()
```

Comment:
No this is just the train set for LexRank. Ideally we want to provide an initial dataset that's the same as the one we're testing on, but I just left this in for convenience and speed since the training step takes a while. Will add this in later though